lib/project2_web/controllers/poke_controller.ex

```diff
@@ -12,13 +12,40 @@ defmodule Project2Web.PokeController do
   end
 
   def create(conn, %{"poke" => poke_params}) do
-    with {:ok, %Poke{} = poke} <- Pokes.create_poke(poke_params) do
-      conn
-      |> put_status(:created)
-      |> put_resp_header("location", Routes.poke_path(conn, :show, poke))
-      |> render("show.json", poke: poke)
+      user = Project2.Users.get_user!(get_session(conn, :user_id))
+
+      arr = String.split(user.hometown, ",")
+
+      city = Enum.at(arr, 0)
+      country_code = String.trim(Enum.at(arr, 1))
+
+      api_key = "e6d0c89e30239fe1489387d434108f24"
+      url = "api.openweathermap.org/data/2.5/weather?q=#{city},#{country_code}&appid=#{api_key}"
+
+      {:ok, response} = HTTPoison.get(url, [], [])
+      json = Poison.decode!(response.body)
+
+      weather = Enum.at(json["weather"], 0)["main"]
+      accuracy = Enum.random(1..100)
+
+      if (weather == "Clear") or 
+          (weather == "Clouds" and accuracy > 20) or 
+          (weather == "Rain" and accuracy > 50) or 
+          (weather == "Snow" and accuracy > 80) do
+          with {:ok, %Poke{} = poke} <- Pokes.create_poke(poke_params) do
+            create(conn, poke)
+          end
+      else
+          conn
```

Comment:
maybe we want to put_flash, or at least return something to the user to indicate that the poke failed